homeassistant/components/light/futurenow.py

```diff
@@ -0,0 +1,140 @@
+        device = {}
+        device['name'] = device_config[CONF_NAME]
+        device['dimmable'] = device_config['dimmable']
+        device['channel'] = channel
+        device['driver'] = config[CONF_DRIVER]
+        device['host'] = config[CONF_HOST]
+        device['port'] = config[CONF_PORT]
+        lights.append(FutureNowLight(device))
+
+    add_devices(lights, True)
+
+
+def to_futurenow_level(level):
+    """Convert the given HASS light level (0-255) to FutureNow (0-100)."""
+    return int((level * 100) / 255)
+
+
+def to_hass_level(level):
+    """Convert the given FutureNow (0-100) light level to HASS (0-255)."""
+    return int((level * 255) / 100)
+
+
+class FutureNowLight(Light):
+    """Representation of an FutureNow light."""
+
+    def __init__(self, device):
+        """Initialize the light."""
+        import pyfnip
+
+        self._name = device['name']
+        self._dimmable = device['dimmable']
+        self._channel = device['channel']
+        self._brightness = None
+        self._state = None
+        self._skip_update = False
+
+        if device['driver'] == CONF_DRIVER_FNIP6X10AD:
+            self._light = pyfnip.FNIP6x2adOutput(device['host'],
+                                                 device['port'],
+                                                 self._channel)
+        if device['driver'] == CONF_DRIVER_FNIP8X10A:
+            self._light = pyfnip.FNIP8x10aOutput(device['host'],
+                                                 device['port'],
+                                                 self._channel)
+
+    @property
+    def name(self):
+        """Return the name of the device if any."""
+        return self._name
+
+    @property
+    def is_on(self):
+        """Return true if device is on."""
+        return self._state
+
+    @property
+    def brightness(self):
+        """Return the brightness of this light between 0..255."""
+        return self._brightness
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        if self._dimmable:
+            return SUPPORT_BRIGHTNESS
+        return 0
+
+    def turn_on(self, **kwargs):
+        """Turn the light on."""
+        level = kwargs.get(ATTR_BRIGHTNESS, 255) if self._dimmable else 255
+        self._light.turn_on(to_futurenow_level(level))
+        self._brightness = level
+        self._state = True
+        self._skip_update = True
+
+    def turn_off(self, **kwargs):
+        """Turn the light off."""
+        self._light.turn_off()
+        self._brightness = 0
+        self._state = False
+        self._skip_update = True
+
+    def update(self):
+        """Fetch new state data for this light."""
+        if self._skip_update:
+            self._skip_update = False
+            return
+
+        state = int(self._light.is_on())
+        if state:
+            self._state = True
+            self._brightness = to_hass_level(state)
```

Comment:
This should work both when state is 0 and above 0. We can move it out of the `if state:` block.